vendor-extra/JatsContentBundle/src/ViewConverter/FrontContentHeaderVisitor.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\JatsContentBundle\ViewConverter;
+
+use FluentDOM\DOM\Document;
+use FluentDOM\DOM\Element;
+use Libero\ViewsBundle\Views\SimplifiedVisitor;
+use Libero\ViewsBundle\Views\View;
+use Libero\ViewsBundle\Views\ViewConverter;
+use Libero\ViewsBundle\Views\ViewConverterVisitor;
+
+final class FrontContentHeaderVisitor implements ViewConverterVisitor
+{
+    use SimplifiedVisitor;
+
+    private $converter;
+
+    public function __construct(ViewConverter $converter)
+    {
+        $this->converter = $converter;
+    }
+
+    protected function doVisit(Element $object, View $view, array &$context = []) : View
+    {
+        /** @var Document $document */
+        $document = $object->ownerDocument;
+        $xpath = $document->xpath();
+        $xpath->registerNamespace('jats', 'http://jats.nlm.nih.gov');
+
+        $title = $xpath->firstOf('jats:article-meta/jats:title-group/jats:article-title', $object);
+
+        if (!$title instanceof Element) {
+            return $view;
+        }
+
+        return $view->withArgument(
+            'contentTitle',
+            $this->converter->convert($title, '@LiberoPatterns/heading.html.twig', $context)->getArguments()
+        );
+    }
+
+    protected function expectedTemplate() : string
+    {
+        return '@LiberoPatterns/content-header.html.twig';
+    }
+
+    protected function expectedElement() : string
+    {
+        return '{http://jats.nlm.nih.gov}front';
+    }
+
+    protected function unexpectedArguments() : array
```

Comment:
this seems to be a consolidated way to avoid accidentally overwriting an existing view argument, but I'm not sure what decides the priority of the Visitors

Reply:
The service tag https://github.com/libero/browser/blob/c7a401521a4dca7fe15bc9b0fae691bebe84e860/vendor-extra/ViewsBundle/src/Resources/config/services.xml#L19 (one for documentation)